blocks/product-details/product-details.js

```diff
@@ -1,26 +1,270 @@
 /* eslint-disable import/no-extraneous-dependencies */
-import { events } from '@dropins/tools/event-bus.js';
-import { initializers } from '@dropins/tools/initializer.js';
+
 import {
   InLineAlert,
   Icon,
+  Button,
   provider as UI,
 } from '@dropins/tools/components.js';
-import * as productApi from '@dropins/storefront-pdp/api.js';
-import { render as productRenderer } from '@dropins/storefront-pdp/render.js';
+import { events } from '@dropins/tools/event-bus.js';
+import { initializers } from '@dropins/tools/initializer.js';
+import * as PDP from '@dropins/storefront-pdp/api.js';
+import { render as PDPProvider } from '@dropins/storefront-pdp/render.js';
 import { addProductsToCart } from '@dropins/storefront-cart/api.js';
-import ProductDetails from '@dropins/storefront-pdp/containers/ProductDetails.js';
+
+// Containers
+import ProductHeader from '@dropins/storefront-pdp/containers/ProductHeader.js';
+import ProductPrice from '@dropins/storefront-pdp/containers/ProductPrice.js';
+import ProductShortDescription from '@dropins/storefront-pdp/containers/ProductShortDescription.js';
+import ProductOptions from '@dropins/storefront-pdp/containers/ProductOptions.js';
+import ProductQuantity from '@dropins/storefront-pdp/containers/ProductQuantity.js';
+import ProductDescription from '@dropins/storefront-pdp/containers/ProductDescription.js';
+import ProductAttributes from '@dropins/storefront-pdp/containers/ProductAttributes.js';
+import ProductGallery from '@dropins/storefront-pdp/containers/ProductGallery.js';
 
 // Libs
 import {
-  getProduct,
-  getSkuFromUrl,
   setJsonLd,
   loadErrorPage, performCatalogServiceQuery, variantsQuery,
+  getSkuFromUrl,
+  getOptionsUIDsFromUrl,
 } from '../../scripts/commerce.js';
-import { getConfigValue } from '../../scripts/configs.js';
 import { fetchPlaceholders } from '../../scripts/aem.js';
+import { getConfigValue } from '../../scripts/configs.js';
+
+export default async function decorate(block) {
+  // Set Fetch Endpoint (Service)
+  PDP.setEndpoint(await getConfigValue('commerce-endpoint'));
+
+  // Set Fetch Headers (Service)
+  PDP.setFetchGraphQlHeaders({
+    'Content-Type': 'application/json',
+    'Magento-Environment-Id': await getConfigValue('commerce-environment-id'),
+    'Magento-Website-Code': await getConfigValue('commerce-website-code'),
+    'Magento-Store-View-Code': await getConfigValue('commerce-store-view-code'),
+    'Magento-Store-Code': await getConfigValue('commerce-store-code'),
+    'Magento-Customer-Group': await getConfigValue('commerce-customer-group'),
+    'x-api-key': await getConfigValue('commerce-x-api-key'),
+  });
+
+  // pre-fetch PDP data
+  const sku = getSkuFromUrl();
+  const optionsUIDs = getOptionsUIDsFromUrl();
+
+  const [product, labels] = await Promise.all([
+    PDP.fetchPDPData(sku, { optionsUIDs }),
+    fetchPlaceholders(),
+  ]);
+
+  // Initialize
+  initializers.register(PDP.initialize, {
+    sku,
+    optionsUIDs,
```

Comment:
You can now pass the optionsUIDs here, but also through the models (as before). What's the difference? Isn't this a bit redundant or potentially cause confusion?

Reply:
It's a way to provide `optionsUIDs` you may want to have pre-selected on the page load without having to pass the whole `initialData`.